### ISSUE TITLE
Integrity not ensured

### ISSUE THREAD

**Issue by owittek** [itzg/discord-github-melder-bot#85]
Hey, thanks for making this bot!

I'm currently working on a bot that has the same purpose (but written in Rust for practice) so I was looking around for already existing solutions. The problem that I'm facing and that seems to also be prevalent in this bot is that the actual integrity of the verification is not ensured. It's possible for a user to just copy the verification link and let another user, who is a contributor, to do the verification for them since there's no direct communication between discord & github.

Not sure if you've already been aware of this or if I just didn't see your solution to this problem within the source code.

**Comment by itzg** [comment]
The link will force the user to go through Github login, so I'm not quite sure what scenario you're concerned about. 

In any case, I'm planning on rewriting this whole thing now that Discord added native linking with GitHub accounts.

**Comment by itzg** [comment]
...and I think I see what you mean  since it doesn't take them through a Discord login. So, yes there's a window where the code could get hijacked but seems like the user would have to go out of their way to share that before it gets used/expired.

**Comment by owittek** [comment]
yeah they would have to share the link before it expires.

How exactly are you planning to rewrite it? Although people can link their github account to discord I assume that they have to also show it on their profile and they might not want to share their github profile on discord just for a single server.

Could you please point me in the right direction?

**Comment by itzg** [comment]
I actually hadn't gotten that far to research how the new APIs expose their GitHub linkage. I was going to assume/require that if people want to link their GitHub then they need to share it accordingly.

**Comment by owittek** [comment]
yep, that's why I assume that a user might not want to share their github with everyone and might find it too cumbersome to hide it again. But I guess you're right about that if they want to be show off being a contributor then this is also required.
